File: lib/src/services/cache_service.dart
class CacheService {
  final Map<String, dynamic> _cache = {};
  final Map<String, int> _ttls = {};

  Future<dynamic> get(String key) async {
    if (_ttls[key] != null && DateTime.now().millisecondsSinceEpoch > _ttls[key]!) {
      delete(key);
      return null;
    }
    return _cache[key];
  }

  Future<void> set(String key, dynamic value, {int ttl = 3600}) async {
    _cache[key] = value;
    _ttls[key] = DateTime.now().millisecondsSinceEpoch + (ttl * 1000);
  }

  Future<bool> delete(String key) async {
    _cache.remove(key);
    _ttls.remove(key);
    return true;
  }
}

File: DataBase/todoapp/App/Services/CacheService.php
<?php
namespace App\Services;

class CacheService {
    private $cache = [];
    private $ttls = [];  // برای نگهداری زمان انقضا

    public function get($key) {
        if (isset($this->ttls[$key]) && time() > $this->ttls[$key]) {
            $this->delete($key);
            return null;
        }
        return $this->cache[$key] ?? null;
    }

    public function set($key, $value, $ttl = 3600) {
        $this->cache[$key] = $value;
        $this->ttls[$key] = time() + $ttl;
    }

    public function delete($key) {
        unset($this->cache[$key], $this->ttls[$key]);
    }
} 